### PR TITLE
992: Hide soft keyboard on unlock.

### LINKED ISSUE #992: Keyboard is not automatically hidden in entry list view when unlocking the app using a passcode
## Steps to reproduce
Have a passcode set in the device.
Lock the app and unlock it
### Expected behavior
The entry list view is shown completely
### Actual behavior
The keyboard hides part of the entries. You have to manually hide it to see the full list

### Device & build information
* Device: Pixel Android 10
* Build version: master 2c445568b30be0432a85f27e6ae822248b6a9ff7

### PR DESCRIPTION
**Pull request by eliserichards**
Fixes #992

## Testing and Review Notes
Hiding the keyboard on `onPause()` for the `LockedFragment`. There's a little bit of a "flash" of the splash screen in between unlocking and the item list. Not sure if we should open a issue to fix that, or ...?

## Screenshots or Videos
autofill_capture_duolingo

## To Do

- [x] double check the original issue to confirm it is fully satisfied
- [x] add testing notes and screenshots in PR description to help guide reviewers
- [x] request the "UX" team perform a design review (if/when applicable)
- [ ] make sure CI builds are passing (e.g.: fix lint and other errors)

### CODE REVIEW

**Comment by changecourse** [comment]
>There's a little bit of a "flash" of the splash screen in between unlocking and the item list. Not sure if we should open a issue to fix that, or ...?

It would be nice to open an additional issue if we're not able to fix that as part of this PR. I had noticed this on 3.0 when unlocking from biometrics, but hadn't had a chance to log.

**Comment by eliserichards** [comment]
Filed #1027.

File: app/src/main/java/mozilla/lockbox/view/ItemListFragment.kt [hunk 1]
```diff
@@ -225,7 +233,10 @@ class ItemListFragment : Fragment(), ItemListView {
         if (!networkErrorVisibility) {
             errorHelper.showNetworkError(view!!)
         } else {
-            errorHelper.hideNetworkError(parent = view!!, child = view!!.refreshContainer.entriesView)
+            errorHelper.hideNetworkError(
+                parent = view!!,
+                child = view!!.refreshContainer.entriesView
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/view/ItemListFragment.kt]: 😻

File: app/src/main/java/mozilla/lockbox/view/LockedFragment.kt [hunk 2]
```diff
@@ -32,11 +34,21 @@ class LockedFragment : Fragment(), LockedView {
...
 
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
         super.onActivityResult(requestCode, resultCode, data)
         _onActivityResult.onNext(Pair(requestCode, resultCode))
     }
+
+    private fun closeKeyboard() {
+        val imm = context?.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+        imm.hideSoftInputFromWindow(view!!.windowToken, 0)
+    }
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/view/LockedFragment.kt]: I wonder whether we should put this into the base `Fragment` class and call it from the base `onPause()` method.

**Pull request merged**